test/vfd_swmr_indep_rw_writer.c

```diff
@@ -0,0 +1,788 @@
+    }
+
+    if ((dset_id = H5Dcreate2(s->file[which], dname, s->dtype, filespace, H5P_DEFAULT, H5P_DEFAULT,
+                              H5P_DEFAULT)) < 0) {
+        HDfprintf(stderr, "H5Dcreate2 failed\n");
+        TEST_ERROR;
+    }
+
+    if (H5Sclose(filespace) < 0) {
+        HDfprintf(stderr, "H5Sclose failed\n");
+        TEST_ERROR;
+    }
+
+    if (!init_matrix(mat, which)) {
+        HDfprintf(stderr, "data initialization failed\n");
+        TEST_ERROR;
+    }
+
+    if (H5Dwrite(dset_id, H5T_NATIVE_UINT32, H5S_ALL, H5S_ALL, H5P_DEFAULT, mat->elt) < 0) {
+        HDfprintf(stderr, "H5Dwrite failed\n");
+        TEST_ERROR;
+    }
+
+    if (H5Dclose(dset_id) < 0) {
+        HDfprintf(stderr, "H5Dclose failed\n");
+        TEST_ERROR;
+    }
+
+    if (s->first_proc)
+        dbgf(1, "Process 0: Successfully write the dataset %s for file %s.\n", dname, s->filename[which]);
+    else
+        dbgf(1, "Process 1: Successfully write the dataset %s for file %s.\n", dname, s->filename[which]);
+
+    return true;
+
+error:
+    H5E_BEGIN_TRY
+    {
+        H5Sclose(filespace);
+        H5Dclose(dset_id);
+    }
+    H5E_END_TRY;
+
+    return false;
+}
+
+static bool
+open_dset(state_t *s)
+{
+
+    char         dname[sizeof("/dataset-?")];
+    hid_t        dset_id;
+    unsigned int i;
+    unsigned int fopen_idx = 0;
+
+    /* The first process reads the dataset from the second process
+     * The second process reads the dataset from the first process
+     */
+    if (s->first_proc) {
+        esnprintf(dname, sizeof(dname), "/dataset-%d", 1);
+        fopen_idx = 1;
+    }
+    else {
+        esnprintf(dname, sizeof(dname), "/dataset-%d", 0);
+        fopen_idx = 0;
+    }
+
+    /* Same as the big dset test, tries to open the dataset repeatedly until successful.  After trying
+     * NUM_ATTEMPTS times without success, report it as a failure
+     */
+    for (i = 0; i < NUM_ATTEMPTS; i++) {
+        H5E_BEGIN_TRY
+        {
+            dset_id = H5Dopen2(s->file[fopen_idx], dname, H5P_DEFAULT);
+        }
+        H5E_END_TRY;
+
+        if (dset_id >= 0)
+            break;
+        else
+            decisleep(s->check_interval);
+    }
+
+    s->r_dsetid = dset_id;
+    if (i == NUM_ATTEMPTS) {
+        HDfprintf(stderr, "dataset opening reachs the maximal number of attempts\n");
+        TEST_ERROR;
+    }
+
+    if (s->first_proc)
+        dbgf(1, "Process 0: Successfully open the dataset %s for file %s.\n", dname, s->filename[fopen_idx]);
+    else
+        dbgf(1, "Process 1: Successfully open the dataset %s for file %s.\n", dname, s->filename[fopen_idx]);
+    return true;
+
+error:
+    return false;
+}
+
+static bool
+vrfy_dset(const state_t *s, mat_t *mat)
+{
+
+    unsigned int i;
+    unsigned int which = 0;
+    herr_t       status;
+
+    /* A failure to read the data may indicate the data isn't ready yet.  Instead of displaying the error
+     * stack, simply return false and let the caller repeat this step.
```

Comment:
Is the comment correct: "...let the caller repeat this step."?

Reply:
Will check

Reply:
Revise the comments to make it clearer.